include/aos/common/crypto.hpp

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_CRYPTO_HPP_
+#define AOS_CRYPTO_HPP_
+
+#include "aos/common/tools/array.hpp"
+#include "aos/common/tools/string.hpp"
+
+#include "aos/common/time.hpp"
+#include "aos/common/types.hpp"
+
+namespace aos {
+namespace crypto {
+
+/**
+ * Certificate public key len.
+ */
+constexpr auto cCertPubKeyLen = AOS_CONFIG_CRYPTO_CERT_PUB_KEY_LEN;
+
+/**
+ * Certificate private key len.
+ */
+constexpr auto cCertPrivKeyLen = AOS_CONFIG_CRYPTO_CERT_PRIV_KEY_LEN;
+
+/**
+ * Certificate serial number max length.
+ */
+constexpr auto cCertSerialNumberLen = AOS_CONFIG_CRYPTO_CERT_SERIAL_LEN;
+
+/**
+ * Certificate issuer name max length.
+ */
+constexpr auto cCertIssuerLen = AOS_CONFIG_CRYPTO_CERT_ISSUER_ID_LEN;
+
+/**
+ * Max length of alternative module name.
+ */
+constexpr auto cDnsNameLen = AOS_CONFIG_CRYPTO_DNS_NAME_LEN;
+
+/**
+ * Max number of alternative names for a module.
+ */
+constexpr auto cAltDnsNamesCount = AOS_CONFIG_CRYPTO_ALT_DNS_NAMES_MAX_COUNT;
+
+/**
+ * Certificate subject size.
+ */
+constexpr auto cCertSubjSize = AOS_CONFIG_CRYPTO_CERT_SUBJECT_SIZE;
+
+/**
+ * Raw CSR max size.
+ */
+constexpr auto cRawCSRSize = AOS_CONFIG_CRYPTO_RAW_CSR_SIZE;
+
+/**
+ * Certificate extra extrnsions max number.
+ */
+constexpr auto cCertExtraExtCount = AOS_CONFIG_CRYPTO_EXTRA_EXTENSIONS_COUNT;
+
+/**
+ * Maximum length of numeric string representing ASN.1 Object Identifier.
+ */
+constexpr auto cASN1ObjIdLen = AOS_CONFIG_CRYPTO_OBJECT_ID_LEN;
+
+/**
+ * Maximum size of a certificate extension value.
+ */
+constexpr auto cCertExtValueSize = AOS_CONFIG_CRYPTO_CERT_EXTENSION_VALUE_SIZE;
+
+/**
+ * Maximum certificate key id size(in bytes).
+ */
+constexpr auto cCertKeyIdSize = AOS_CONFIG_CRYPTO_CERT_KEY_ID_SIZE;
+
+/**
+ * General certificate private key type.
+ */
+class PrivateKey {
+public:
+    /**
+     * Returns public part of a private key.
+     */
+    const Array<uint8_t>& GetPublic() const { return mPublicKey; }
+
+    /**
+     * Returns private part of a private key.
+     */
+    const Array<uint8_t>& GetPrivate() const { return mPrivateKey; }
```

Comment:
Again, private key should not be exposed as public API in either case. So remove this method as well as cCertPrivKeyLen.